test-two/src/index.js

```diff
@@ -1,19 +1,44 @@
 import { log } from "./helpers/logger";
 import { getUsersForOrganisation } from "./services/github";
 
+
+
 const baseUrl = "https://api.github.com/";
 const organisationId = "google";
-const gitHubApiAuthToken = ""; // add your GitHub API OAuth key here to increase request limit
+const gitHubApiAuthToken = process.env.ACCESSS_TOKEN; // add your GitHub API OAuth key here to increase request limit
+
+const MAX_PER_PAGE = 100; // max users per page is 100 ; default is 30
+var page = 1; // start from first page
+const allUsers =[]; // initialise the array to store all users in it
 
 export const go = async () => {
   try {
-    const users = await getUsersForOrganisation(
+    var users = await getUsersForOrganisation(
       baseUrl,
       gitHubApiAuthToken,
-      organisationId
+      organisationId,
+      MAX_PER_PAGE,
+      page
     );
-
-    const output = users.reduce((result, user) => {
+    var length = MAX_PER_PAGE
+    while(length === MAX_PER_PAGE){
```

Comment:
interesting way of looping through and knowing when you're on the last page

Reply:
Do you think there's a way to write this as an async function so that we can fetch multiple pages as a time. Say there were hundreds of pages and we wanted to speed up this fetching of the list.

Reply:
Yes I think can use async-await inside the function which has increased the speed of fetching all users dramatically 
here is my code below for it

` const output = await allUsers.reduce(async (result, user) =>  {
       console.log(result);
       return (`${await result}Username: ${user.login}\n`);
    }, "");`